addons/stock_account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,334 @@
+                        'product_id': product.id,
+                        'description': _('Product value manually modified (from %s to %s)') % (price_at_date, pph[4]),
+                        'value': value,
+                        'quantity': 0,
+                        'old_product_price_history_id': pph[0],
+                        'create_date': pph[3],
+                        'create_uid': pph[5] or 1,
+                        'write_uid': pph[6] or 1,
+                        'write_date': pph[7],
+                    }
+                    _create_stock_valuation_layer(env, [vals])
+                    std_price_update[company.id, product.id] = pph[4]
+
+
+def _get_price_unit(move, std_price_update):
+    """ Returns the unit price to value this stock move """
+    move.ensure_one()
+    price_unit = move.price_unit
+    # If the move is a return, use the original move's price unit.
+    returned_svl = move.env['stock.valuation.layer'].search([
+        ('stock_move_id', '=', move.origin_returned_move_id.id),
+    ], order='create_date desc, id desc', limit=1)
+    if move.origin_returned_move_id and returned_svl:
+        price_unit = returned_svl.unit_cost
+    if not move.company_id.currency_id.is_zero(price_unit):
+        return price_unit
+    return std_price_update.get(move.company_id.id, move.product_id.id) or move.product_id.standard_price
+
+
+def _create_stock_valuation_layer(env, vals_list):
+    for vals in vals_list:
+        if 'account_move_id' not in vals:
+            vals['account_move_id'] = _get_related_account_move(env, vals).id or None
+        columns = vals.keys()
+        query = """
+            INSERT INTO stock_valuation_layer ({})
+            VALUES({})
+        """.format(', '.join(columns), ", ".join(["%({})s".format(col) for col in columns]))
+        openupgrade.logged_query(env.cr, query, vals)
+
+
+def _get_related_account_move(env, svl_vals):
+    """ Return Account move related to Stock Valuation Layer"""
+    domain = []
+    if svl_vals.get('stock_move_id'):
+        domain = expression.AND([domain, [('stock_move_id', '=', svl_vals['stock_move_id'])]])
+    if svl_vals.get('old_product_price_history_id'):
+        env.cr.execute("""
+            SELECT create_date
+            FROM product_price_history
+            WHERE id = %s
+        """, (svl_vals['old_product_price_history_id'],))
+        create_date = env.cr.fetchone()[0]
+        domain = expression.AND([domain, [('create_date', '=', create_date)]])
+    if not domain:
+        return env['account.move']
+    account_moves = env['account.move'].search(domain)
+    if len(account_moves) > 1:
+        return env['account.move']
+    return account_moves
+
+
+def _compute_value_svl(env, product, company):
+    # We call this function because the product doesn't have in cache correct value
+    # when we create SVL by SQL
+    # use sql instead of ORM because it saves ~30% time
+    openupgrade.logged_query(env.cr, """
```

Comment:
Oh, I mistook. We can use `env.cr.execute(` in order to not spam in logs

Reply:
done